app/controllers/katello/api/registry/registry_proxies_controller.rb

```diff
@@ -268,6 +383,27 @@ def finish_upload_blob
       head pulp_response.code
     end
 
+    def push_manifest
+      headers = translated_headers_for_proxy
+      headers['Content-Type'] = request.headers['Content-Type'] if request.headers['Content-Type']
+      body = @_request.body.read
+      pulp_response = Resources::Registry::Proxy.put(@_request.fullpath, body, headers)
+      pulp_response.headers.each do |key, value|
+        response.header[key.to_s] = value
+      end
+
+      # if manifest push was successful, trigger content indexing on the repo
+      if 200 <= pulp_response.code && pulp_response.code <= 299
+        unless @product.root_repositories.where(label: @container_name).first.library_instance.index_content
```

Comment:
So you were pushing an image that was manifest only so it only hit the push_manifest controller method? I suppose it would make sense to make the repo creation logic on all four podman-related API endpoints, probably as a before action.

Reply:
yeah it seems only the manifest was getting pushed since all the blobs already existed in the Pulp repo

Reply:
This issue has been resolved. Repos are now created whenever they do not already exist.